### PR TITLE
stateUpgrades Documentation

### PR DESCRIPTION
**Pull request by darioush**
To be merged along with https://github.com/ava-labs/subnet-evm/pull/549
Provides documentation for `stateUpgrades`

### CODE REVIEW

File: docs/subnets/customize-a-subnet.md [hunk 1]
```diff
@@ -1061,3 +1061,75 @@ If `allowFeeRecipients` or `RewardManager` precompile is enabled on the Subnet,
 doesn't specify a "feeRecipient", the fees will be burned in blocks it produces.
 
 :::
+
+## Network Upgrades: State Modification
```
Comment (aaronbuchwald) [inline docs/subnets/customize-a-subnet.md:1065]:
```suggestion
## Network Upgrades: State Upgrades
```

File: docs/subnets/customize-a-subnet.md [hunk 2]
```diff
@@ -1061,3 +1061,75 @@ If `allowFeeRecipients` or `RewardManager` precompile is enabled on the Subnet,
...
+
+SubnetEVM allows the network operators to specify a modification to state that will take place
```
Comment (aaronbuchwald) [inline docs/subnets/customize-a-subnet.md:1067]:
```suggestion
Subnet-EVM allows the network operators to specify a modification to state that will take place
```

File: docs/subnets/customize-a-subnet.md [hunk 3]
```diff
@@ -1061,3 +1061,75 @@ If `allowFeeRecipients` or `RewardManager` precompile is enabled on the Subnet,
...
+    {
+      "blockTimestamp": 1678239000,
+      "accounts": {
+        "0x71562b71999873DB5b286dF957af199Ec94617F7": {
+          "code": "0xdeadbeef"
+        },
+        "0xFF00000000000000000000000000000000000000": {
+          "balanceChange": "0x64",
+          "storage": {
+            "0x0000000000000000000000000000000000000000000000000000000000001234": "0x0000000000000000000000000000000000000000000000000000000000006666"
```
Comment (aaronbuchwald) [inline docs/subnets/customize-a-subnet.md:1127]: Can we use a real address here?

File: docs/subnets/customize-a-subnet.md [hunk 4]
```diff
@@ -1061,3 +1061,75 @@ If `allowFeeRecipients` or `RewardManager` precompile is enabled on the Subnet,
...
+- Sets the storage slot `0x1234` to the value `0x6666` for the account at `0xFF00000000000000000000000000000000000000`.
+
+```json
+{
+  "stateUpgrades": [
+    {
+      "blockTimestamp": 1678239000,
+      "accounts": {
+        "0x71562b71999873DB5b286dF957af199Ec94617F7": {
+          "code": "0xdeadbeef"
```
Comment (aaronbuchwald) [inline docs/subnets/customize-a-subnet.md:1122]: Can we use an example runtime code byte slice here?

File: docs/subnets/customize-a-subnet.md [hunk 5]
```diff
@@ -1061,3 +1061,75 @@ If `allowFeeRecipients` or `RewardManager` precompile is enabled on the Subnet,
...
+`stateUpgrades` can be specified along with `precompileUpgrades` or by itself.
+
+The following three state modifications are supported:
+
+- `balanceChange`: adds a specified amount to the balance of a given account. This amount can be
+  specified as hex or decimal and must be positive.
+- `storage`: modifies the specified storage slots to the specified values. Keys and values must
+  be 32 bytes specified in hex, with a `0x` prefix.
+- `code`: modifies the code of a contract at the provided address to the specified code. The
+  code must _only_ be the runtime portion of a code.
```
Comment (aaronbuchwald) [inline docs/subnets/customize-a-subnet.md:1097]:
```suggestion
- `code`: modifies the code stored in the specified account. The
  code must _only_ be the runtime portion of a code. 
```

**Review by anusha-ctrl — APPROVED** [review]

**Review by yulin-dong — APPROVED** [review]

**Pull request merged**